portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -37,13 +37,28 @@
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String slimit = request.getParameter("limit");
+    int limit;
+
+    try {
+      limit = Integer.parseInt(slimit);
+      if(limit<0)
+        limit = Integer.MAX_VALUE;
+    } catch (Exception e) {
+      limit = Integer.MAX_VALUE;
```

Comment:
Do you really want to ignore this exception here? If this is actually an error, you should throw an exception.